packages/data-driver-postgres/src/query/index.ts

```diff
@@ -4,21 +4,31 @@ import { select } from './select.js';
 import type { ParameterizedSQLStatement } from '@directus/data-sql';
 import { limit } from './limit.js';
 import { offset } from './offset.js';
+import { orderBy } from './orderBy.js';
 
 /**
  * Constructs an actual PostgreSQL query statement from a given abstract SQL query.
  *
+ * @remarks
+ * To create a PostgreSQL statement each part is constructed in a separate function.
+ * In those functions it will be checked if the part should actually be created.
+ * If not, the functions return null.
+ *
  * @param query - The abstract SQL statement
- * @returns An actual SQL with parameters
+ * @returns An actual SQL query with parameters
  */
 export function constructSqlQuery(query: AbstractSqlQuery): ParameterizedSQLStatement {
 	const base = [select(query), from(query)].join(' ');
 
+	const orderByPart = orderBy(query);
+
 	const limitPart = limit(query);
 	const offsetPart = offset(query);
 
+	const actualParts = [base, orderByPart, limitPart, offsetPart].filter((p) => p !== null);
+
 	return {
-		statement: `${base} ${limitPart} ${offsetPart}`.trimEnd() + ';',
+		statement: actualParts.join(' ') + ';',
```

Comment:
Not entirely happy with this part for the following reasons:
- `join(' ')` is used two times for same purpose (to build `base`  and final `statement`)
- `actualParts` is bit of a vague naming
  - Maybe `statementParts` or `statementComponents`
- `base` is checked for `null` although it's always a string
  - However, fine for me to keep that for better readability (see below)

So it could look something like this

```ts
export function constructSqlQuery(query: AbstractSqlQuery): ParameterizedSQLStatement {
	const orderByPart = orderBy(query);

	const limitPart = limit(query);
	const offsetPart = offset(query);

	const statementParts =
		[select(query), from(query), ...[orderByPart, limitPart, offsetPart].filter((p) => p !== null)];

	return {
		statement: `${statementParts.join(' ')};`,
		parameters: query.parameters,
	};
}
```

or

```ts
export function constructSqlQuery(query: AbstractSqlQuery): ParameterizedSQLStatement {
	const orderByPart = orderBy(query);

	const limitPart = limit(query);
	const offsetPart = offset(query);

	const statementParts = [select(query), from(query), orderByPart, limitPart, offsetPart];

	return {
		statement: `${statementParts.filter((p) => p !== null).join(' ')};`,
		parameters: query.parameters,
	};
}
```

or even

```ts
export function constructSqlQuery(query: AbstractSqlQuery): ParameterizedSQLStatement {
	const statementParts = [select(query), from(query), orderBy(query), limit(query), offset(query)];

	return {
		statement: `${statementParts.filter((p) => p !== null).join(' ')};`,
		parameters: query.parameters,
	};
}
```

could consider using something like

```ts
export function constructSqlQuery(query: AbstractSqlQuery): ParameterizedSQLStatement {
	const statementParts = [select, from, orderBy, limit, offset];

	const statement = `${statementParts
		.map((part) => part(query))
		.filter((p) => p !== null)
		.join(' ')};`;

	return {
		statement,
		parameters: query.parameters,
	};
}
```

Reply:
you're totally right. the fourth one looks very good to me 👀  let me commit that

Reply:
Great, thanks 🚀